Packs/Legacy/Scripts/ZipFile/ZipFile.py

```diff
@@ -2,100 +2,117 @@
-
-with open(zipName, 'rb') as f:
-    file_data = f.read()
-
-demisto.results(fileResult(zipName, file_data))
-results = [
-    {
-        'Type': entryTypes['note'],
-        'ContentsFormat': formats['json'],
-        'Contents': {'ZippedFiles': zipName},
-        'EntryContext': {
-            'ZippedFiles': zipName,
-            'ZipFile.ZippedFile': zipName,
-            'File(val.EntryID=="' + fileEntryID + '").zipped': True
-        },
-        'ReadableContentsFormat': formats['markdown'],
-        'HumanReadable': tableToMarkdown('Zipped Files', [{'original name': fileCurrentName, 'zipped file': zipName}])
-    }]
-
-demisto.results(results)
+ESCAPE_CHARACTERS = r'[/\<>"|?*]'
+
+
+def escape_illegal_characters_in_file_name(file_name):
+    if file_name:
+        file_name = re.sub(ESCAPE_CHARACTERS, '-', file_name)
+        file_name = re.sub('-+', '-', file_name)
+
+    return file_name
+
+
+def main():
+    try:  # in order to support compression of the file
+        compression = zipfile.ZIP_DEFLATED
+
+    except Exception:
+        compression = zipfile.ZIP_STORED
+
+    zipName = None
+    filePath = None
+    password = None
+    fileEntryID = demisto.args().get('entryID')
+
+    if 'zipName' in demisto.args().keys():
+        zipName = escape_illegal_characters_in_file_name(demisto.args().get('zipName')) + '.zip'
+
+    if 'password' in demisto.args().keys():
+        password = demisto.args().get('password')
+
+    if not fileEntryID:
+        demisto.results({
+            'Type': entryTypes['error'],
+            'ContentsFormat': formats['text'],
+            'Contents': 'You must set an entryID when using the zip script'
+        })
+        sys.exit(0)
+
+    res = demisto.executeCommand('getFilePath', {'id': fileEntryID})
+
+    if res[0]['Type'] == entryTypes['error']:
+        demisto.results({
+            'Type': entryTypes['error'],
+            'ContentsFormat': formats['text'],
+            'Contents': 'Failed to get the file path for entry: ' + fileEntryID + ' the error message was '
+                        + res[0]['Contents']
+        })
+        sys.exit(0)
+
+    filePath = res[0]['Contents']['path']
+    fileCurrentName = escape_illegal_characters_in_file_name(res[0]['Contents']['name'])
+
+    if not zipName:
+        zipName = fileCurrentName + '.zip'
+
+    if not isfile(filePath):  # in case that the user will send a directory
+        demisto.results({
+            'Type': entryTypes['error'],
+            'ContentsFormat': formats['text'],
+            'Contents': fileEntryID + 'is not a file. Please recheck your input.'
+        })
+        sys.exit(0)
+
+    # copying the file to current location
+    shutil.copy(filePath, fileCurrentName)
+    # zipping the file
+    if password:
+        pyminizip.compress(fileCurrentName, zipName, password, 5)
+
+    else:
+        zf = zipfile.ZipFile(zipName, mode='w')
+        try:
+            zf.write(fileCurrentName, compress_type=compression)
```

Comment:
If you changed the fileCurrentName above in relation to how it is "written" in the system (`res[0]['Contents']['name']`) - how does the zip know to fetch the file here?

Reply:
`shutil.copy(filePath, fileCurrentName)` copies the file from the Demisto entry id to the `fileCurrentName` which was altered prior to it.
So, in the `zf.write` the file name is the right now.